specifyweb/frontend/js_src/lib/components/QueryBuilder/CheckReadAccess.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { commonText } from '../../localization/common';
+import { queryText } from '../../localization/query';
+import { RA } from '../../utils/types';
+import { Ul } from '../Atoms';
+import { SerializedResource } from '../DataModel/helperTypes';
+import { SpQuery, Tables } from '../DataModel/types';
+import { Dialog } from '../Molecules/Dialog';
+import { TableIcon } from '../Molecules/TableIcon';
+import { hasTablePermission } from '../Permissions/helpers';
+import { QueryFieldSpec } from './fieldSpec';
+
+export function CheckReadAccess({
+  query,
+}: {
+  readonly query: SerializedResource<SpQuery>;
+}): JSX.Element | null {
+  const [noAccessTables, setNoAccessTables] = React.useState<RA<keyof Tables>>(
+    []
+  );
+
+  function accessTables(query: SerializedResource<SpQuery>): RA<keyof Tables> {
+    const info = query.fields.flatMap((field) => {
+      const fieldSpec = QueryFieldSpec.fromStringId(
+        field.stringId,
+        field.isRelFld ?? false
+      );
+      const relatedTable = fieldSpec.joinPath.map(({ model }) => model.name);
```

Comment:
The `relatedTable` variable name is incorrect since it is assigned to an array. Also, you really don't need that variable and can just return right away

Reply:
Can you replace `fieldSpec.joinPath.map(({ model }) => model.name);` with `fieldSpec.joinPath.map((field)=>field.isRelationship ? field.relatedModel.name : field.model.name)`?

This way, fields like `Accession -> (formatted)` would be checked correctly